components/org.wso2.carbon.identity.cloud.listener.claim.onprem/src/main/java/org/wso2/carbon/identity/cloud/listener/claim/onprem/CloudClaimManagerListener.java

```diff
@@ -0,0 +1,196 @@
+import org.wso2.carbon.identity.cloud.listener.claim.onprem.security.SecurityTokenBuilder;
+import org.wso2.carbon.identity.cloud.listener.claim.onprem.util.EndpointUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.user.api.ClaimManager;
+import org.wso2.carbon.user.api.ClaimMapping;
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.api.Tenant;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.common.AbstractClaimManagerListener;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.io.IOException;
+import java.security.Key;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+/**
+ * Custom claim manager listener for secondary userstores.
+ */
+public class CloudClaimManagerListener extends AbstractClaimManagerListener {
+    private static Log log = LogFactory.getLog(CloudClaimManagerListener.class);
+    private int tenantId;
+    private static final String ENDPOINT = "EndPointURL";
+    private HttpClient httpClient;
+    private static Map<Integer, Key> privateKeys = new ConcurrentHashMap<>();
+    private ClaimManager claimManager;
+    private UserStoreManager secondaryUserStoreManager;
+    private String tenantDomain;
+
+    @Override
+    public boolean getAttributeName(String domainName, String claimURI)
+            throws org.wso2.carbon.user.core.UserStoreException {
+        try {
+            tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+            Tenant tenant = ClaimListenerComponentHolder.getInstance().getRealmService()
+                    .getTenantManager().getTenant(tenantId);
+            if (tenant != null) {
+                tenantDomain = ClaimListenerComponentHolder.getInstance().getRealmService()
+                        .getTenantManager().getTenant(tenantId).getDomain();
+                TenantDomainClaimCacheEntry cacheEntry = getTenantDomainReferenceFromCache(
+                        getTenantDomainCacheKey(domainName, tenantDomain));
+                if (cacheEntry == null) {
+                    secondaryUserStoreManager = ((UserStoreManager) (ClaimListenerComponentHolder.getInstance()
+                            .getRealmService().getTenantUserRealm(tenantId).getUserStoreManager()))
+                            .getSecondaryUserStoreManager(domainName);
+
+                    if (secondaryUserStoreManager != null) {
+                        claimManager = secondaryUserStoreManager.getClaimManager();
+                        ClaimMapping[] claimMappings = claimManager.getAllClaimMappings();
+                        GetMethod getMethod = new GetMethod(EndpointUtil.getClaimAttributeRetrievalEndpoint(getHostName()));
+
+                        if (this.httpClient == null) {
+                            this.httpClient = new HttpClient();
+                        }
```

Comment:
Why not initialize the http client for the constructor or in the instance variable it self